lua/precognition/compat.lua

```diff
@@ -0,0 +1,10 @@
+local M = {}
+
+function M.inlay_hints_enabled(t)
+    if vim.lsp and vim.lsp.inlay_hint and vim.lsp.inlay_hint.is_enabled then
```

Comment:
Couldn't you use `has("nvim-0.10")` here?

Reply:
If this isn't a big deal then this PR is good to merge imo :)

Reply:
I need to rebase and will check the field name. I know the lsp API changed a few times in 0.10. I can't remember which released

I will check docs tomorrow